Sources/PackLib/Packer.swift

```diff
@@ -91,70 +134,79 @@ public struct Packer: Sendable {
                 }
             }
-            if let iconPath = plan.iconPath {
-                group.addTask {
-                    try await packFileToRoot(srcName: iconPath)
-                }
-            }
+        }
+        if let iconPath = product.iconPath {
             group.addTask {
-                try await packFile(srcName: "\(plan.product)-App", dstName: plan.product)
+                try await packFileToRoot(srcName: iconPath)
             }
-            group.addTask {
-                var info = plan.infoPlist
-
-                if let iconPath = plan.iconPath {
-                    let iconName = URL(fileURLWithPath: iconPath).deletingPathExtension().lastPathComponent
-                    info["CFBundleIconFile"] = iconName
-                }
+        }
+        group.addTask {
+            try await packFile(srcName: product.targetName, dstName: product.product)
+        }
+        group.addTask {
+            var info = product.infoPlist
 
-                let infoPath = outputURL.appendingPathComponent("Info.plist")
-                let encodedPlist = try PropertyListSerialization.data(
-                    fromPropertyList: info,
-                    format: .xml,
-                    options: 0
-                )
-                try encodedPlist.write(to: infoPath)
-            }
-            while !group.isEmpty {
-                do {
-                    try await group.next()
-                } catch is CancellationError {
-                    // continue
-                } catch {
-                    group.cancelAll()
-                    throw error
-                }
+            if let iconPath = product.iconPath {
+                let iconName = URL(fileURLWithPath: iconPath).deletingPathExtension().lastPathComponent
+                info["CFBundleIconFile"] = iconName
             }
+
+            let infoPath = outputURL.appendingPathComponent("Info.plist")
+            let encodedPlist = try PropertyListSerialization.data(
+                fromPropertyList: info,
+                format: .xml,
+                options: 0
+            )
+            try encodedPlist.write(to: infoPath)
         }
+    }
+}
 
-        let dest = URL(fileURLWithPath: "xtool")
-            .appendingPathComponent(output.url.lastPathComponent)
-        try? FileManager.default.removeItem(at: dest)
-        try output.persist(at: dest)
-        return dest
+extension Plan.Product {
+    fileprivate var swiftPMFlags: String {
+        switch self.type {
+        case .application: ""
+        case .appExtension: """
+        ,
+        cSettings: [.unsafeFlags(["-fapplication-extension"])],
+        swiftSettings: [.unsafeFlags(["-Xfrontend", "-application-extension"])],
+        linkerSettings: [
+            // Link to Foundation framework which implements the _NSExtensionMain entrypoint
+            .linkedFramework("Foundation"),
+            .unsafeFlags([
+                "-Xlinker", "-application_extension",
```

Comment:
did the flags I’d set not work for you? It’s generally better to pass flags to the driver and let it figure out what to forward to the frontend/linker; in this case swift-driver seems to know what to do when you give it `-application-extension`